lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -838,36 +838,67 @@ class ADContext {
   SILPassManager &getPassManager() const { return passManager; }
   Lowering::TypeConverter &getTypeConverter() { return module.Types; }
 
-  SmallVectorImpl<DifferentiableFunctionInst *> &
-  getDifferentiableFunctionInsts() {
-    return differentiableFunctionInsts;
+  /// Returns true if there are `DifferentiableFunctionInst` to process.
+  bool hasDifferentiableFunctionInsts() const {
```

Comment:
How about:
```suggestion
  bool isDifferentiableFunctionInstWorklistEmpty() const {
```
Mentioning "worklist" clarifies where the `DifferentiableFunctionInst` came from.
With this new name, the method body should be the inverse: `return differentiableFunctionInsts.empty();`.